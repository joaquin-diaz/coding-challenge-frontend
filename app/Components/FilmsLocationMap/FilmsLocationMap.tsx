import * as React from 'react';

import Map from '../Map';
import Marker from '../Marker';

import Film from '../../types/Film';

interface FilmsLocationMapProps {
  films: Array<Film>;
  onMarkerClick: (film: Film) => void;
}

const FilmsLocationMap: React.FC<FilmsLocationMapProps> = props => {
  const renderMarker = (film: Film, key: number) => {
    return (
      <Marker
        film={film}
        key={key}
        lat={film.coordinates.lat}
        lng={film.coordinates.lng}
        onClick={props.onMarkerClick}
      />
    );
  };
  const sanFranciscoCenter = {
    lat: 37.7900484,
    lng: -122.4021149,
  };

  return (
    <div>
      <Map center={sanFranciscoCenter}>{props.films.map(renderMarker)}</Map>
    </div>
  );
};

export default FilmsLocationMap;
