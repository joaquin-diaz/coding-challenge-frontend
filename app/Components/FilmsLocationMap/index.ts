import FilmsLocationMap from './FilmsLocationMap';

export default FilmsLocationMap;
