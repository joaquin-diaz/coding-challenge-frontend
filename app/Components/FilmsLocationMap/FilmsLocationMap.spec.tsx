import * as React from 'react';

import FilmsLocationMap from './FilmsLocationMap';
import { shallow } from 'enzyme';
import { fakeFilm } from '../../utils/test.fixtures';

const noop = () => {};

describe('<FilmsLocationMap />', () => {
  it('should render with markers', () => {
    const wrapper = shallow(<FilmsLocationMap films={[fakeFilm, fakeFilm]} onMarkerClick={noop} />);
    const markers = wrapper.find('Marker');

    expect(wrapper).toMatchSnapshot();
    expect(markers.length).toEqual(2);
    expect(markers.at(0).prop('film')).toEqual(fakeFilm);
    expect(markers.at(0).prop('lat')).toEqual(fakeFilm.coordinates.lat);
    expect(markers.at(0).prop('lng')).toEqual(fakeFilm.coordinates.lng);
    expect(markers.at(0).prop('onClick')).toEqual(noop);
  });
});
