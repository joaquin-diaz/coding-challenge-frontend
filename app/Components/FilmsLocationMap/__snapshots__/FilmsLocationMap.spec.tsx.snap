// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilmsLocationMap /> should render with markers 1`] = `
<div>
  <Map
    center={
      Object {
        "lat": 37.7900484,
        "lng": -122.4021149,
      }
    }
  >
    <Marker
      film={
        Object {
          "actor1": "Siddarth",
          "actor2": "Nithya Menon",
          "actor3": "Priya Anand",
          "address": "Epic Roasthouse (399 Embarcadero)",
          "coordinates": Object {
            "lat": 123,
            "lng": 456,
          },
          "director": "Jayendra",
          "productionCompany": "SPI Cinemas",
          "releaseYear": "2011",
          "title": "180",
          "writer": "Umarji Anuradha, Jayendra, Aarthi Sriram, & Suba ",
        }
      }
      key="0"
      lat={123}
      lng={456}
      onClick={[Function]}
    />
    <Marker
      film={
        Object {
          "actor1": "Siddarth",
          "actor2": "Nithya Menon",
          "actor3": "Priya Anand",
          "address": "Epic Roasthouse (399 Embarcadero)",
          "coordinates": Object {
            "lat": 123,
            "lng": 456,
          },
          "director": "Jayendra",
          "productionCompany": "SPI Cinemas",
          "releaseYear": "2011",
          "title": "180",
          "writer": "Umarji Anuradha, Jayendra, Aarthi Sriram, & Suba ",
        }
      }
      key="1"
      lat={123}
      lng={456}
      onClick={[Function]}
    />
  </Map>
</div>
`;
